### PR TITLE
[WIP] rust-sdk-python-bindings

### PR DESCRIPTION
**Pull request by gnunicorn**

### CODE REVIEW

File: _projects/rust-sdk-python-bindings [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ - python
+ - Rust
+ - FFI
+ - Nio
+---
+This is an awesome project idea.
+
+#### Description
+
+[matrix-nio](https://github.com/poljar/matrix-nio) the python library to connect to matrix, lacks some important features - most notably e2ee-support - that the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk) supports. This projects want to create python bindings for the Rust SDK and integrate it in the nio library.
```
Comment (Cadair) [inline _projects/rust-sdk-python-bindings:27]: but nio does have e2ee support?

Comment (gnunicorn) [reply _projects/rust-sdk-python-bindings:27]: sorry, meant `cross-signing support`...

File: _projects/rust-sdk-python-bindings [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+name: Matrix-Rust-SDK-Python
+desc: Building Python/nio bindings for using the matrix rust sdk
+requirements:
+# Student requirements:
+ - Wokred with Rust FFI
```
Comment (gnunicorn) [inline _projects/rust-sdk-python-bindings:6]:
```suggestion
 - Worked with Rust FFI
```

File: _projects/rust-sdk-python-bindings (hunk 1 above)
Comment (gnunicorn) [reply _projects/rust-sdk-python-bindings:27]:
```suggestion
[matrix-nio](https://github.com/poljar/matrix-nio) the python library to connect to matrix, lacks some important features - most notably cross-signing-support - that the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk) supports. This projects want to create python bindings for the Rust SDK and integrate it in the nio library.
```

File: _projects/rust-sdk-python-bindings [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ - python
+ - Rust
+ - FFI
+ - Nio
+---
+This is an awesome project idea.
+
+#### Description
+
+[matrix-nio](https://github.com/poljar/matrix-nio) the python library to connect to matrix, lacks some important features - most notably cross-signing-support - that the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk) supports. This projects want to create python bindings for the Rust SDK and integrate it in the nio library.
```
Comment (poljar) [inline _projects/rust-sdk-python-bindings:27]: Let's mention that a state store is also lacking from nio and the rust-sdk would take care of that.

>  [matrix-nio](https://github.com/poljar/matrix-nio), a Python Matrix client library, lacks some important features, up-to-date cryptography, the ability to store state, support for reactions, etc. All of those missing features are supported by the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk).

> This project aims to create Python bindings for the Rust SDK and integrate them in the matrix-nio library. matrix-nio already contains multiple `Client` classes for different use-cases, the `AsyncClient` class API is similar to the Rust SDK API. The project should introduce a new `Client` class to `matrix-nio` which would replace the `AsyncClient` class.

File: _projects/rust-sdk-python-bindings [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+name: Matrix-Rust-SDK-Python
+desc: Building Python/nio bindings for using the matrix rust sdk
+requirements:
+# Student requirements:
+ - Worked with Rust FFI
+ - Familiar with building Python Bindings
+difficulty: low
```
Comment (poljar) [inline _projects/rust-sdk-python-bindings:8]: I would probably rate this medium, you need to be proficient in two languages and there's quite a lot to do, bindings, docs, integration into an existing project...

File: _projects/rust-sdk-python-bindings [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+name: Matrix-Rust-SDK-Python
+desc: Building Python/nio bindings for using the matrix rust sdk
```
Comment (poljar) [inline _projects/rust-sdk-python-bindings:3]:
```suggestion
desc: Building Python bindings for using the Matrix Rust SDK
```

**Review by poljar — CHANGES_REQUESTED** [review]
Some small nits and added some details to the description.

File: _projects/rust-sdk-python-bindings (hunk 4 above)
Comment (gnunicorn) [reply _projects/rust-sdk-python-bindings:8]:
```suggestion
difficulty: medium
```

File: _projects/rust-sdk-python-bindings (hunk 3 above)
Comment (gnunicorn) [reply _projects/rust-sdk-python-bindings:27]:
```suggestion
[matrix-nio](https://github.com/poljar/matrix-nio), a Python Matrix client library, lacks some important features, up-to-date cryptography, the ability to store state, support for reactions, etc. All of those missing features are supported by the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk).

This project aims to create Python bindings for the Rust SDK and integrate them in the matrix-nio library. matrix-nio already contains multiple `Client` classes for different use-cases, the `AsyncClient` class API is similar to the Rust SDK API. The project should introduce a new `Client` class to `matrix-nio` which would replace the `AsyncClient` class.
```

**Pull request merged**